profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -57,6 +59,49 @@ unsigned long long dd_inside_wrapped_functions()
     return functions_entered_counter;
 }
 
+__attribute__((constructor))
+void initLibrary(void) {
+    // If crashtracking is enabled, check the value of DOTNET_DbgEnableMiniDump
+    // If set, set DD_TRACE_CRASH_HANDLER_PASSTHROUGH to indicate dd-dotnet that it should call createdump
+    // If not set, set it to 1 so that .NET calls createdump in case of crash
+    // (and we will redirect the call to dd-dotnet)
+    char* crashHandler = getenv("DD_TRACE_CRASH_HANDLER");
+
+    if (crashHandler != NULL && crashHandler[0] != '\0')
+    {
+        char* enableMiniDump = getenv("DOTNET_DbgEnableMiniDump");
+
+        if (enableMiniDump == NULL)
+        {
+            enableMiniDump = getenv("COMPlus_DbgEnableMiniDump");
+        }
+
+        if (enableMiniDump != NULL && enableMiniDump[0] == '1')
+        {
+            // If DOTNET_DbgEnableMiniDump is set, the crash handler should call createdump when done
+            char* passthrough = getenv("DD_TRACE_CRASH_HANDLER_PASSTHROUGH");
+
+            if (passthrough == NULL || passthrough[0] == '\0')
+            {
+                // If passthrough is already set, don't override it
+                // This handles the case when, for example, the user calls dotnet run
+                //  - dotnet run sets DOTNET_DbgEnableMiniDump=1
+                //  - dotnet then launches the target app
+                //  - the target app thinks DOTNET_DbgEnableMiniDump has been set by the user and enables passthrough
+                setenv("DD_TRACE_CRASH_HANDLER_PASSTHROUGH", "1", 1);
+            }
+        }
+        else
+        {
+            // If DOTNET_DbgEnableMiniDump is not set, we set it so that the crash handler is called,
+            // but we instruct it to not call createdump afterwards
+            setenv("COMPlus_DbgEnableMiniDump", "1", 1);
+            setenv("DOTNET_DbgEnableMiniDump", "1", 1);
+            setenv("DD_TRACE_CRASH_HANDLER_PASSTHROUGH", "0", 1);
```

Comment:
Where do we look at this env var `DD_TRACE_CRASH_HANDLER_PASSTHROUGH` ?

Reply:
The value is then used by dd-dotnet (in `CreatedumpCommand.cs` to know whether it should call createdump or not.